addons/account/models/account_move.py

```diff
@@ -1234,7 +1234,20 @@ def _compute_tax_totals(self):
                             handle_price_include=False,
                             extra_context={'_extra_grouping_key_': 'epd'},
                         ))
-                move.tax_totals = self.env['account.tax']._prepare_tax_totals(**kwargs)
+
+                amount_untaxed = abs(move.amount_untaxed_signed)
+                amount_total = abs(move.amount_total_signed)
+                currency = move.currency_id or move.journal_id.currency_id or move.company_id.currency_id
+                if move.state == 'posted':
+                    move.tax_totals = {**self.env['account.tax']._prepare_tax_totals(**kwargs),
+                                       'amount_untaxed': amount_untaxed,
+                                       'amount_total': amount_total,
+                                       'formatted_amount_total': formatLang(self.env, amount_total, currency_obj=currency),
+                                       'formatted_amount_untaxed': formatLang(self.env, amount_untaxed, currency_obj=currency)
+                                       }
```

Comment:
What about early payment discount ("EPD") and all that stuff? Does it work here?